### PR TITLE
fix: Catalog: add COUNT OVER to UNION query to get total_rows correctly

### LINKED ISSUE #274: Add `total` count for catalog endpoint when fetching with no NETWORK specified
When the network is not specified, the endpoint performns a UNION ALL query to both subgraph, consolidating the results later on. In order to know the total of rows (sum of both subgraph total), the UNION query needs to be wrapped in a subquery and a COUNT OVER() needs to be added before applying the LIMIT and OFFSET (so we get the right amount).

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #274

### CODE REVIEW

File: src/ports/catalog/utils.ts [hunk 1]
```diff
@@ -79,14 +76,20 @@ const getMultiNetworkQuery = (
       network: network as Network,
     })
   )
-  const unionQuery = SQL``
+
+  // The following code wraps the UNION query in a subquery so we can get the total count of items before applying the limit and offset
```
Comment (kevinszuchet) [inline src/ports/catalog/utils.ts:80]: 👏👏👏

File: src/tests/ports/catalog.spec.ts [hunk 2]
```diff
@@ -244,6 +244,10 @@ test('catalog component', function () {
           ],
           total: 1,
         })
+        console.log(
+          'dbClientQueryMock.mock.calls: ',
+          dbClientQueryMock.mock.calls
+        )
```
Comment (kevinszuchet) [inline src/tests/ports/catalog.spec.ts:250]: Would you mind removing this `console.log` if not relevant 🙏 ?

Comment (juanmahidalgo) [reply src/tests/ports/catalog.spec.ts]: sure, done!

File: src/ports/catalog/utils.ts [hunk 3]
```diff
@@ -79,14 +76,20 @@ const getMultiNetworkQuery = (
...
+  const unionQuery = SQL`SELECT *, COUNT(*) OVER() as total FROM (\n`
   queries.forEach((query, index) => {
     unionQuery.append(query)
     if (queries[index + 1]) {
       unionQuery.append(SQL`\n UNION ALL \n`)
     }
   })
-  addQuerySortAndPagination(unionQuery, filters)
+  addQuerySort(unionQuery, filters)
+  unionQuery.append(SQL`\n) as tem \n`)
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/ports/catalog/utils.ts]: I thought it was a reserved word, but seems it's available to use, changed to `temp`!

**Pull request merged**